### ISSUE TITLE
[Bug]: Confirmation Mode not working

### ISSUE THREAD

**Issue by amirshawn** [All-Hands-AI/OpenHands#5608]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

I updated to the newest version of open hands and enabled confirmation mode.  I no longer get a prompt to approve an action.  It basically gets stuck in the first step.  I'm not sure if I'm doing something wrong but previous versions worked as expected.

### OpenHands Installation

Development workflow

### OpenHands Version

0.15.2

### Operating System

MacOS

**Comment by amirshawn** [comment]
Not being able to use confirmation mode is making it very difficult to use this system.  It goes rogue way too often.  Is there a demonstration about how to use open hands with github?

**Comment by enyst** [comment]
I set this as frontend to begin with, because there doesn't seem to be anything visible in the frontend for the user to Confirm.

**Comment by mamoodi** [comment]
Upgrading the severity because it is fully broken.

**Comment by amirshawn** [comment]
Thank you @mamoodi one thing I don't quite understand is why we can't toggle confirmation mode on and off without restarting the session?  If this can be achieve it definitely should be because using pause still allows the agent to keep doing other things without recognizing the message you sent it in between.

**Comment by mamoodi** [comment]
@amirshawn yeah that's probably because it gets bundled into "settings" like changing the agent. It shouldn't be necessary. You can open another issue for that if you'd like.

**Comment by mamoodi** [comment]
This seems fixed in main. Awaiting release.

**Comment by mamoodi** [comment]
This is fixed in 0.22. Closing.
